src/components/test/Checkbox-test.js

```diff
@@ -0,0 +1,106 @@
+import { mount } from 'enzyme';
+
+import { checkAccessibility } from '../../../test/util/accessibility';
+
+import { LabeledCheckbox, Checkbox } from '../Checkbox';
+
+describe('Checkbox', () => {
+  function createComponent(props = {}) {
+    return mount(<Checkbox name="my-checkbox" {...props} />);
+  }
+
+  it('inputRef points to the input element', () => {
+    const inputRef = {};
+    createComponent({ inputRef });
+    assert.isTrue(inputRef.current instanceof HTMLInputElement);
+  });
+
+  it('invokes onClick callback on click', () => {
+    const inputRef = {};
+    const onClick = sinon.stub();
+    createComponent({ inputRef, onClick });
+
+    inputRef.current.click();
+    assert.calledOnce(onClick);
+  });
+});
+
+describe('LabeledCheckbox', () => {
+  function createComponent(props = {}) {
+    return mount(
+      <div>
+        {/** wrapped on a div to pass the a11y test */}
+        <LabeledCheckbox name="my-checkbox" {...props}>
+          {props.children || 'My Action'}
+        </LabeledCheckbox>
+      </div>
+    );
+  }
+
+  it('contains a label after the input element (default position)', () => {
+    const inputRef = {};
+    const wrapper = createComponent({ inputRef });
+    const labelElement = wrapper.find('label');
+    assert.equal(labelElement.text(), 'My Action');
+    assert.equal(inputRef.current.nextElementSibling.tagName, 'LABEL');
+  });
+
+  it('contains a label before the input element ', () => {
```

Comment:
I think this test description could use an additional clause, to explain in which cases the label would come before the input (i.e. the `position` prop is set to `before`)...

Reply:
Addressed, thanks.